### PR TITLE
Don't remove existential quantification (#339)

### PR DESCRIPTION
**Pull request by imuli**
- also don't remove context from data declarations

Fixes: https://github.com/haskell/stylish-haskell/issues/339

### CODE REVIEW

**Comment by EncodePanda** [comment]
@jaspervdj I don't seem to have the rights to re-run the CI, could you possibly do that?

**Comment by EncodePanda** [comment]
@imuli do you want to pair up to push this PR forward?

**Comment by imuli** [comment]
I have limited personal bandwith at the moment, but I will attempt to find some. I can at least rebase onto the current `main` branch if that's helpful.

**Comment by imuli** [comment]
Looks like the CI on macOS is still having problems.

**Comment by EncodePanda** [comment]
I've just fixed mac cache. Can you rerun?

**Comment by EncodePanda** [comment]
By rerun I mean rebase one more time, please :)

**Comment by cdornan** [comment]
It would be awesome if we could get this patch released.

**Comment by EncodePanda** [comment]
@cdornan I've bumped the CI so the tests are passing now. I will evaluate the PR over the weekend and if all is good we will get it in.

**Comment by EncodePanda** [comment]
@imuli could we add 

```
case60 :: Assertion
case60 = assertSnippet (step defaultConfig)
    [ "data Foo = forall a . Bar a" ]
    [ "data Foo = forall a. Bar a" ]
```

for completeness?

File: tests/Language/Haskell/Stylish/Step/Data/Tests.hs [hunk 1]
```diff
@@ -1291,6 +1292,14 @@ case58 = expected @=? testStep (step sameIndentStyle) input
 
     expected = input
 
+-- | Don't remove existential quantification
+--
+-- See https://github.com/jaspervdj/stylish-haskell/issues/339
```
Comment (EncodePanda) [inline tests/Language/Haskell/Stylish/Step/Data/Tests.hs:1297]: please update the comment to https://github.com/jaspervdj/haskell/issues/339

**Review by EncodePanda — COMMENTED** [review]
@imuli this is amazing work! thank you so much for the contribution!

Please add the missing test `case60` (just copy-paste from the comment) and update the link to the issue in the comment.

One more time, thank so much for this :)

**Review by EncodePanda — APPROVED** [review]
LTGM

**Pull request merged**